bundles/org.openhab.binding.gce/src/main/java/org/openhab/binding/gce/internal/handler/Ipx800v3Handler.java

```diff
@@ -0,0 +1,282 @@
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.gce.internal.config.AnalogInputConfiguration;
+import org.openhab.binding.gce.internal.config.CounterConfiguration;
+import org.openhab.binding.gce.internal.config.DigitalInputConfiguration;
+import org.openhab.binding.gce.internal.config.Ipx800Configuration;
+import org.openhab.binding.gce.internal.config.RelayOutputConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ipx800v3Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class Ipx800v3Handler extends BaseThingHandler implements Ipx800EventListener {
+    private static final String PROPERTY_SEPARATOR = "-";
+
+    private final Logger logger = LoggerFactory.getLogger(Ipx800v3Handler.class);
+
+    private @NonNullByDefault({}) Ipx800Configuration configuration;
+    private @NonNullByDefault({}) Ipx800DeviceConnector connector;
+    private Optional<Ipx800MessageParser> parser = Optional.empty();
+
+    private final Map<String, @Nullable PortData> portDatas = new HashMap<>();
+
+    private class LongPressEvaluator implements Runnable {
+        private final ZonedDateTime referenceTime;
+        private final String port;
+        private final String eventChannelId;
+
+        public LongPressEvaluator(Channel channel, String port, PortData portData) {
+            this.referenceTime = portData.getTimestamp();
+            this.port = port;
+            this.eventChannelId = channel.getUID().getId() + PROPERTY_SEPARATOR + CHANNEL_TYPE_PUSH_BUTTON_TRIGGER;
+        }
+
+        @Override
+        public void run() {
+            PortData currentData = portDatas.get(port);
+            if (currentData != null && currentData.getValue() == 1 && currentData.getTimestamp() == referenceTime) {
+                triggerChannel(eventChannelId, EVENT_LONG_PRESS);
+            }
+        }
+    }
+
+    public Ipx800v3Handler(Thing thing) {
+        super(thing);
+        logger.debug("Create a IPX800 Handler for thing '{}'", getThing().getUID());
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
After the thing status is set to OFFLINE, the framework starts to work with this Thing: `handleCommand()` and `dispose()` can be invoked (concurrently). You might want to remove it as you set the state at the end of the method.